app/src/main/java/com/android/sample/ui/authentication/SignUp.kt

```diff
@@ -0,0 +1,183 @@
+package com.android.sample.ui.authentication
+
+import android.content.Context
+import android.util.Log
+import android.util.Patterns
+import android.widget.Toast
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Column
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.padding
+import androidx.compose.foundation.layout.size
+import androidx.compose.material3.Button
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.material3.TextButton
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import com.android.sample.R
+import com.android.sample.ui.navigation.NavigationActions
+import com.android.sample.ui.navigation.Screen
+import com.google.firebase.auth.ktx.auth
+import com.google.firebase.ktx.Firebase
+
+fun isValidEmail(email: String): Boolean {
+  return Patterns.EMAIL_ADDRESS.matcher(email).matches()
+}
+
+@Composable
+fun SignUpScreen(navigationActions: NavigationActions) {
+  val context = LocalContext.current
+  val emailState = remember { mutableStateOf("") }
+  val passwordState = remember { mutableStateOf("") }
+  val emailErrorState = remember {
+    mutableStateOf<String?>(null)
+  } // State for email validation error
+  val passwordErrorState = remember {
+    mutableStateOf<String?>(null)
+  } // State for password validation error
```

Comment:
Again, for these, could be better to have a real state data class with Sucess and Error